app/src/main/java/com/firebase/uidemo/database/ChatIndexActivity.java

```diff
@@ -34,12 +35,13 @@ public void onClick(View v) {
 
     @Override
     protected FirebaseRecyclerAdapter<Chat, ChatHolder> getAdapter() {
-        return new FirebaseIndexRecyclerAdapter<Chat, ChatHolder>(
-                Chat.class,
+        return new FirebaseRecyclerAdapter<Chat, ChatHolder>(
+                new FirebaseIndexArray<>(
```

Comment:
@samtstern Ok, I decided to go with outside developer happiness instead of ours and added a whole bunch of overloads.